node/pkg/fetcher/fetcher.go

```diff
@@ -1 +1,142 @@
 package fetcher
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"bisonai.com/orakl/node/pkg/bus"
+	"bisonai.com/orakl/node/pkg/db"
+	"bisonai.com/orakl/node/pkg/utils"
+)
+
+const FETCHER_FREQUENCY = 2 * time.Second
+
+func NewFetcher(bus *bus.MessageBus) *Fetcher {
+	return &Fetcher{
+		Adapters: make([]AdapterDetail, 0),
+		Bus:      bus,
+	}
+}
+
+func (f *Fetcher) Run(ctx context.Context) error {
+	err := f.initialize(ctx)
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(FETCHER_FREQUENCY)
+	go func() {
+		for range ticker.C {
```

Comment:
If I understand it correctly, all adapters are triggered at the same time. Previously, we had all adapters executed in independent recurrent operations that could overlap but did not have to. We should strive for desynchronized fetcher execution so the load to 3rd party API does not come in batches and is distributed in between fetcher executions. We don't have to change it in this PR, but we should at least make a PR for this functionality.

Reply:
Martin, I have tried to add logic to run fetch iterations evenly among fetcher frequency which is 2 seconds at the moment.

I have defined duration for each execution by dividing fetcher frequency into length of adapter length. 

for example if we have 2 adapters, it'll be 2 / 2 which is 1.
during the iteration, it'll sleep if execution time hasn't exceeded 1 second

Reply:
@nick-bisonai I see. I think it is a bit dangerous implementation in general, not just in due to the last update. It would be much better if
* fetching is executed independently of other adapters
* fetching of the same adapter cannot overlap

Reply:
I understand the condition you're explaining but I can't quite take a grasp of how I should implement it
unless there is a trigger for each adapters and user manually run each adapters

Reply:
thank you for the suggestion, I'll make an update

Reply:
I've made a commit to deal with this issue, please feel free to leave any comments.
https://github.com/Bisonai/orakl/pull/1153/commits/f0eea4f1fc8ef18a6549f3070ac4a9b26a434cfb